test/no-array-reduce.js

```diff
@@ -12,6 +12,8 @@ test({
 		'a[b.reduce]()',
 		'a(b.reduce)',
 		'a.reduce()',
+		'a?.reduce()',
+		'a.reduce?.()',
```

Comment:
It doesn't make sense  to allow optional call.

Reply:
Added it because I saw it in `no-array-sort` and `no-array-reverse` 🤷‍♂️
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/test/no-array-sort.js#L14
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/test/no-array-reverse.js#L14

Reply:
I mean `a?.reduce()` should be **invalid**, but `a.reduce?.()` should be **valid**.

Reply:
Sorry, I comment this in a misleading place, I mean L112

Reply:
I don't understand, sorry :(
Why one should be valid and another invalid? I thought this rule finds usages of `reduce`, and reports it (except trivial callback cases, if corresponding option is provided). Why reporting of `a?.reduce()` should be different to `a.reduce?.()` and `a.reduce()`?
Or `array.reduce((str, item) => str += item, "")` different to `array?.reduce((str, item) => str += item, "")` and `array.reduce?.((str, item) => str += item, "")` if we talk about L112

Reply:
Because it's doesn't make sense to check `Array#reduce`, if user using `foo.reduce?.()` then `foo` can be a `non-array` object.

Explained in this commit [`8240f4f` (#2721)](https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2721/commits/8240f4ff9433625de2184f57726bb0a083772294)